en/6-persistence/README2.md

```diff
@@ -0,0 +1,505 @@
+          else
+          {
+              RequestChangeView();
+              return false;
+          }
+      }
+      return true;
+  }
+
+```
+
+
+#### 0x40 - Accounts
+NEO uses this collection to track user balance, but without tracking UTXO.
+
+#### 0x44 - (Unspent) Coins collections
+A transaction is considered "unspent" if this it is not referenced as input by some transaction. The node keeps track of these unspent coins to validate transactions but doesn't group it by account, meaning it can determine if a coin is spendable, but it cannot get all spendable coins for a specific account. The role of tracking the spendable coins by account is done by the wallet or an indexer service like [neoscan](neoscan.io) or [neotracker](neotracker.io).
+
+This collection is used to determine if the coin is spendable. Here is how it's done in C#:
+
+``` CSharp
+public static bool IsDoubleSpend(this IPersistence persistence, Transaction tx)
+{
+  if (tx.Inputs.Length == 0) return false;
+    foreach (var group in tx.Inputs.GroupBy(p => p.PrevHash))
+    {
+      UnspentCoinState state = persistence.UnspentCoins.TryGet(group.Key);
+      if (state == null) return true;
+      if (group.Any(p => p.PrevIndex >= state.Items.Length || state.Items[p.PrevIndex].HasFlag(CoinState.Spent)))
+        return true;
+      }
+  return false;
+}
+```
+
+Note that at this moment, we are not checking the memory pool. This is done before the `IsDoubleSpend` check.
+
+#### 0x45 - Spent Coins
+NEO tracks the spent coins in order to allow users to `claim GAS`. The claimable GAS is calculated based on interval of blocks between the block where the transaction was created to the moment it is spent.   
+This means that, to be able to Claim GAS, you need to use that transaction. It is common to send it to yourself to "unlock" the claimable GAS.
+
+``` CSharp
+public Dictionary<ushort, SpentCoin> GetUnclaimed(UInt256 hash)
+        {
+    TransactionState tx_state = Transactions.TryGet(hash);
+    if (tx_state == null) return null;
+    SpentCoinState coin_state = SpentCoins.TryGet(hash);
+    if (coin_state != null)
+    {
+        return coin_state.Items.ToDictionary(p => p.Key, p => new SpentCoin
+        {
+            Output = tx_state.Transaction.Outputs[p.Key],
+            StartHeight = tx_state.BlockIndex,
+            EndHeight = p.Value
+        });
+    }
+    else
+    {
+        return new Dictionary<ushort, SpentCoin>();
+    }
+}
+
+```
+
+#### 0x48 - Validators
+This collection is used to verify a block, since it is required to know the validators public keys in order to validate the multiple signatures contained in a block.  
+
+```CSharp
+UInt160[] IVerifiable.GetScriptHashesForVerifying(Snapshot snapshot)
+{
+  ECPoint[] validators = snapshot.GetValidators();
+  if (validators.Length <= ValidatorIndex)
+  throw new InvalidOperationException();
+  return new[] { Contract.CreateSignatureRedeemScript(validators[ValidatorIndex]).ToScriptHash() };
+}
+```
+
```

Comment:
Maybe, we replace by this code is better: https://github.com/neo-project/neo/blob/master-2.x/neo/Consensus/ConsensusContext.cs#L395
As `Validators` is mainly for electing the consensus nodes.  The `NextConsensus` field in block, points to the consesus nodes in the next round.  more details: https://docs.neo.org/developerguide/en/articles/consensus/vote_validator.html